sopel/plugin.py

```diff
@@ -71,6 +89,221 @@
+    request capabilities if they are available. You can request more than one
+    at a time, which will make for one single request.
+
+    The handler must follow this interface::
+
+        from sopel import plugin
+        from sopel.bot import SopelWrapper
+
+        @plugin.capability('example/cap-name')
+        def capability_handler(
+            cap_req: Tuple[str, ...],
+            bot: SopelWrapper,
+            acknowledged: bool,
+        ) -> plugin.CapabilityNegotiation:
+            if acknowledged:
+                # do something if acknowledged
+                # i.e.
+                # activate a plugin's feature
+                pass
+            else:
+                # do something else if not
+                # i.e. use a fallback mechanism
+                # or deactivate a plugin's feature if needed
+                pass
+
+            # always return if Sopel can send "CAP END" (DONE)
+            # or if the plugin must notify the bot for that later (CONTINUE)
+            return CapabilityNegotiation.DONE
+
+    .. note::
+
+        Due to how Capability Negotiation works, a request will be acknowledged
+        or denied all at once. This means that this may succeed::
+
+            @plugin.capability('away-notify')
+
+        But this may not::
+
+            @plugin.capability('away-notify', 'example/incompatible-cap')
+
+        Even though the ``away-notify`` capability is available and can be
+        enabled, the second ``CAP REQ`` will be denied because the server won't
+        acknowledge a request that contains an incompatible capability.
+
+        In that case, if you don't need both at the same time, you should use
+        two different handlers::
+
+            @plugin.capability('away-notify')
+            def cap_away_notify(cap_req, bot, ack):
+                # handle away-notify acknowledgement
+
+            @plugin.capability('example/incompatible-cap')
+            def cap_example_incompatible_cap(cap_req, bot, ack):
+                # handle example/incompatible-cap acknowledgement
+                # or, most probably, lack thereof
+
+        This will allow the server to acknowledge or deny each capability
+        independently.
+
+    .. warning::
+
+        A function cannot be decorated more than once by this decorator, as
+        the result is an instance of :class:`capability`.
+
+        If you want to handle a ``CAP`` message without requesting the
+        capability, you should use the :func:`event` decorator instead.
+
+    .. warning::
+
+        The list of ``cap_req`` is limited in size to prevent the bot from
+        separating the ``CAP REQ`` in multiple lines as the bot does not know
+        how to call back the capability handler upon receiving the multi-line
+        ``ACK * REQ``.
+
+    .. seealso::
+
+        The IRCv3 specification on `Client Capability Negotiation`__.
+
+    .. __: https://ircv3.net/specs/extensions/capability-negotiation.html
```

Comment:
```suggestion
    .. __: https://ircv3.net/specs/extensions/capability-negotiation
```

The site never generates URLs ending in `.html`. (I should probably propose that they add redirects to the "pretty" versions… Shouldn't be too hard; IIRC they use Netlify, like us.)